### PR TITLE
[FileBrowser] - fixed credential disclosure when adding network location

### PR DESCRIPTION
**Pull request by Memphiz**
This fixes the credential disclosure reported in this forum thread:

http://forum.kodi.tv/showthread.php?tid=255009

It also fixes showing the old source after editing it until changing the directory by calling the proper update method.

Not sure who can review this - maybe @xhaggi who did something with dialogs at one point?

### CODE REVIEW

File: xbmc/dialogs/GUIDialogFileBrowser.cpp [hunk 1]
```diff
@@ -961,8 +962,9 @@ bool CGUIDialogFileBrowser::OnPopupMenu(int iItem)
         m_rootDir.SetMask("/");
         m_browsingForFolders = 1;
         m_addNetworkShareEnabled = true;
-        m_selectedPath = newPath;
-        Open();
+        m_selectedPath = url.GetWithoutUserDetails();
+        Update(m_Directory->GetPath());
```
Comment (xhaggi) [inline xbmc/dialogs/GUIDialogFileBrowser.cpp]: we only need to call `Refresh(true)` here

Comment (xhaggi) [inline xbmc/dialogs/GUIDialogFileBrowser.cpp]: ahh sorry `Refresh()` is only part of the `CGUIMediaWindow`

**Comment by xhaggi** [comment]
looks good, but please adjust the comment at https://github.com/xbmc/xbmc/pull/8768/files#diff-c6bd626928fad2d4b6bab10cc4003049R960

`Refresh the dialog` or something similar.

**Comment by Memphiz** [comment]
done jenkins build and merge

**Comment by stefansaraev** [comment]
win32 build fail unrelated.

**Comment by xhaggi** [comment]
could you do a backport please

**Comment by Memphiz** [comment]
backport done ...

**Comment by xhaggi** [comment]
thanks

**Pull request merged**